hs_core/views/utils.py

```diff
@@ -684,6 +704,7 @@ def move_or_rename_file_or_folder(user, res_id, src_path, tgt_path):
     src_file_dir = os.path.dirname(src_full_path)
 
     # ensure the target_full_path contains the file name to be moved or renamed to
+    # TODO: incorrect logic: 'and' is incorrect; either of these is sufficient.
     if src_file_dir != tgt_file_dir and tgt_file_name != src_file_name:
         tgt_full_path = os.path.join(tgt_full_path, src_file_name)
```

Comment:
@alvacouch Regarding your "TODO" comment, I can see the confusion here and should have added more comments to explain/clarify the logic here. This logic was added later to resolve an issue involved when dragging a file into a different directory, in which case, the ```src_file_name``` has to be appended to ``tgt_full_path```. If the pass in parameter ```tgt_full_path``` already has ```src_file_name``` included, in which case, the condition ```tgt_file_name != src_file_name``` would be false, the path join statement is not needed. Another case is when a file name is renamed, in which case the first condition ```src_file_dir != tgt_file_dir``` would be false, and the path join statement is not needed. This path join statement is only needed when both conditions are true which corresponds to the case when users drag a file into a different directory but only has the target directory included, and iRODS imv command requires a data object file name in the target parameter, hence the path join is needed. Does this clarify your confusion here? If so, can you add comments as needed to clarify confusion here? If not, let me know the specifics you think this logic is incorrect so that we can discuss and fix logic if needed.